### ISSUE TITLE
`with-persp-buffer-list` not working

### ISSUE THREAD

**Issue by nandryshak** [Bad-ptr/persp-mode.el#14]
I'm trying to use this macro with `next`/`previous-buffer`, like this:

    (global-set-key (kbd "<right>") (lambda ()
                                        (interactive)
                                        (with-persp-buffer-list () (next-buffer))))

But it's switching to buffers that aren't part of the current perspective. Is this broken? Or am I doing something wrong?

**Comment by Bad-ptr** [comment]
No, it works.)
But the `next-buffer` uses the `window-prev-buffer` function to get a list of buffers.
And I think it is a good idea to filter previous buffer lists for all windows when switching perspective... so I will implement this shortly.

For now you could do add to your config something like this:
```scheme
(add-hook 'persp-activated-hook
                 #'(lambda ()
                     (mapc #'(lambda (w)
                               (set-window-prev-buffers
                                w
                                (delete-if-not #'(lambda (b)
                                                   (memq b
                                                    (safe-persp-buffers (get-frame-persp))))
                                 (window-prev-buffers w))))
                           (window-list))))
```
Then restart emacs (or execute this code(using M-x ielm, or with eval-region or another method), then switch to another perspective and switch back), and now your binding must be working.

**Comment by Bad-ptr** [comment]
This is possibly fixed by a89c9c18140302052a69182210219e9147dd5202.
